### ISSUE TITLE
2.12a18 fails to launch over NullPointerException

### ISSUE THREAD

**Issue by rathsky** [ankidroid/Anki-Android#6568]
Was reviewing with a17, Playstore updated to a18 in the background, now launching ends with the "Collection not opened" error screen. Logcat:

```
…20.748 32104 32104 I AnkiDroid: Launching DeckPicker
…20.748  1036  8186 I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10008000 cmp=com.ichi2.anki/.DeckPicker} from uid 10432
…20.749  1036  8186 W ActivityTaskManager: Activity is launching as a new task, so cancelling activity result.
…20.756  1036  4441 W ActivityTaskManager: Duplicate finish request for ActivityRecord{bc9bd92 u0 com.ichi2.anki/.IntentHandler t2754 f}
…20.770 32104 32104 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@2542bd2
…20.775 32104 32104 I AnkiDroid: User has permissions to access collection
…20.776 32104 32104 I AnkiDroid: Begin openCollection: /storage/emulated/0/AnkiDroid/collection.anki2
…20.824 32104 32104 I AnkiDroid: End openCollection: /storage/emulated/0/AnkiDroid/collection.anki2
…20.825 32104 32104 I AnkiDroid: AnkiActivity::onCreate - DeckPicker
…20.828  2243  2243 D ImageWallpaper: wallpaper visibility changes to: false
…20.878 32104 32104 I AnkiDroid: colOpen: true
…20.879 32104 32104 I AnkiDroid: No startup screens required
…20.883 32104 32104 I AnkiDroid: AnkiActivity::onStart - DeckPicker
…20.890 32104 32104 I AnkiDroid: AnkiActivity::onResume - DeckPicker
…20.959  1036  1219 I ActivityTaskManager: Displayed com.ichi2.anki/.DeckPicker: +238ms
…20.981  2748  2748 I GoogleInputMethodService: GoogleInputMethodService.onFinishInput():3138 
…20.981  2748  2748 I GoogleInputMethodService: GoogleInputMethodService.onStartInput():1808 
…21.041 32104 32190 E AnkiDroid: CollectionTask/ doInBackgroundLoadDeckCounts - error
…21.041 32104 32190 E AnkiDroid: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.ListIterator java.util.List.listIterator()' on a null object reference
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._groupChildrenMain(SchedV2.java:3)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._groupChildrenMain(SchedV2.java:13)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._groupChildrenMain(SchedV2.java:1)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._groupChildren(SchedV2.java:2)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.java:3)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackgroundLoadDeckCounts(CollectionTask.java:3)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:43)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
…21.041 32104 32190 E AnkiDroid:     at android.os.AsyncTask$3.call(AsyncTask.java:378)
…21.041 32104 32190 E AnkiDroid:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
…21.041 32104 32190 E AnkiDroid:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
…21.041 32104 32190 E AnkiDroid:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
…21.041 32104 32190 E AnkiDroid:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
…21.041 32104 32190 E AnkiDroid:     at java.lang.Thread.run(Thread.java:919)
…21.041 32104 32190 E AnkiDroid: 
…21.041 32104 32190 E AnkiDroid: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.ListIterator java.util.List.listIterator()' on a null object reference
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._groupChildrenMain(SchedV2.java:3)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._groupChildrenMain(SchedV2.java:13)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._groupChildrenMain(SchedV2.java:1)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2._groupChildren(SchedV2.java:2)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.libanki.sched.SchedV2.deckDueTree(SchedV2.java:3)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackgroundLoadDeckCounts(CollectionTask.java:3)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:43)
…21.041 32104 32190 E AnkiDroid:     at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
…21.041 32104 32190 E AnkiDroid:     at android.os.AsyncTask$3.call(AsyncTask.java:378)
…21.041 32104 32190 E AnkiDroid:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
…21.041 32104 32190 E AnkiDroid:     at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
…21.041 32104 32190 E AnkiDroid:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
…21.041 32104 32190 E AnkiDroid:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
…21.041 32104 32190 E AnkiDroid:     at java.lang.Thread.run(Thread.java:919)
…21.041 32104 32104 I AnkiDroid: Updating deck list UI
…21.042 32104 32104 E AnkiDroid: DeckPicker/ null result loading deck counts
…21.057 32104 32104 I AnkiDroid: Handling Message: CollectionLoadErrorDialog

```

[NB/edit: Never mind my prior comment here about no longer being able to downgrade, I forgot that _pm install -r -d_ is still a thing.]

**Comment by Arthur-Milchior** [comment]
Thank you for this very quick feedback. I just pushed a correction to https://github.com/ankidroid/Anki-Android/pull/6569 
I am extremely sorry for your time lost and don't get how I did such a silly mistake

**Comment by rathsky** [comment]
Thank you for the quick fix. No need to be sorry for anything: you're doing the work here, I'm the one to be grateful for that.

**Comment by Arthur-Milchior** [comment]
Still sorry for @mikehardy who will lose time with one update to do quickly to alpha

**Comment by mikehardy** [comment]
Nah, my release process is super automated now, as @rathsky said I also appreciate the work == progress. There will be problems because I am *on purpose* integrating change quickly as an intentional balancing act to increase velocity at the cost of what appears to be maybe one "oops" quick-fix a month. In my mind, a totally worthwhile tradeoff. Anyway, solution is pending merge, will be fixed (we think) in 2.12alpha19 in about 30mins to 4 hours depending entirely on how long google takes to chew on it
